arangod/Cluster/MaintenanceWorker.cpp

```diff
@@ -94,6 +94,12 @@ void MaintenanceWorker::nextState(bool actionMore) {
                      || FAILED == _curAction->getState()))
   {
     actionMore = false;
+
+    auto const& description = _curAction->describe();
+    if (description.has("shardVersion")) {
+      TRI_ASSERT(description.has("shard"));
```

Comment:
You could add an assertion to test whether `description["shardVersion"]` equals `_feature.shardVersion(description["shard"])`

(Olny if it is expected to be that way)

Reply:
will rewrite as described by @matthewvon